### ISSUE TITLE
[Performance]: Over 2 minutes for upgrade-interactive resolution phase

### ISSUE THREAD

**Issue by ildella** [yarnpkg/berry#5972]
### Self-service

- [ ] I'd be willing to implement a fix

### Describe the bug

Since almost 10 days, I : 

1. run `yarn upgrade-interactive`
2. select one or more packages to upgrade and press ENTER

Normally, it did take a few seconds in the resolution phase. Now it gets apparently stuck, and finally it resolves in over 2 minutes. 

This happens on all my yarn projects. Since at least November 7. 

![Screenshot from 2023-11-13 14-54-29](https://github.com/yarnpkg/berry/assets/66707/dba16a3f-b2ee-4804-b859-fab47e95eb63)

### To reproduce

Already outlined in the bug description.

### Environment

```shell
System:
    OS: Linux 6.5 Ubuntu 23.10 23.10 (Mantic Minotaur)
    CPU: (8) x64 11th Gen Intel(R) Core(TM) i5-1135G7 @ 2.40GHz
  Binaries:
    Node: 20.9.0 - /tmp/xfs-343f3ec0/node
    Yarn: 4.0.1 - /tmp/xfs-343f3ec0/yarn
    npm: 10.2.3 - ~/n/bin/npm
    bun: 1.0.10 - ~/.bun/bin/bun
```

**Comment by lpic10** [comment]
looks like: https://github.com/yarnpkg/berry/issues/4327 https://github.com/yarnpkg/berry/issues/5167

**Comment by arcanis** [comment]
Can you try the 4.0.2 (literally just released)? It includes a workaround to avoid paying the `getReport` cost.

**Comment by ildella** [comment]
4.0.2 did it!

**Comment by ildella** [comment]
Fixed in 4.0.2

**Comment by tcoopman** [comment]
For me, yarn 4.0.2 doesn't fix the issue
